README
Example Grammar for Julius
---------------------------------

In order to use VoxForge's acoustic models with Julius, you have to tell it
what words you want it to recognize and it what context each of those can
be found. You can do this by writting two files, a .grammar and a .term one,
and generating .dfa and .dict files out of those. This file will explain you
briefly how to achieve this.

== QUICK START ==

First, copy the example files to a directory where you can write, like your
home folder, and uncompress the gziped files:

  mkdir ~/julius-grammar
  cp /usr/share/doc/julius-voxforge/examples/* ~/julius-grammar
  cd ~/julius-grammar
  gunzip * 2>&1 | grep -v ignored

Now let's run the following command (available from package "julius"):

  mkdfa sample

This will generate the files "sample.dfa", "sample.dict" and "sample.term" out
of the "sample.grammar" and "sample.voca" files. Once we have those, we can run
Julius with the following command to try the example grammar out:

  julius -input mic -C julian.jconf

Note that only some sentences are recognized, like for example "DIAL ONE TWO"
or "PHONE STEVE". See the next part to learn how to add more words.

In case your recognition rate is near zero, try replacing "mic" in the
command above with "oss", "alsa" or "esd".

== CREATING OUR OWN GRAMMAR ==

Now open up the files "sample.voca" and "sample.grammar" and have a look at
them. The first one defines some categories (lines starting with "%") and lists
some words, together with their phonetic representation. The .grammar file,
which may look a bit more confusing, defines the context in which each word
category can appear.

Feel free to add new words to the existing categories in sample.voca or even
create new ones (adding at least one rule mentioning them to sample.grammar),
but don't forget to write their phonetic representation next to them. You can
get a list of English words and their corresponding phonetic representation
from ftp://svr-ftp.eng.cam.ac.uk/pub/comp.speech/dictionaries/beep.tar.gz.

About the .grammar file, you need to understand that the lines starting with
"S :" are those which define the possible sentences and that they are formed
by one or more category or variables names, between "NS_B " and " NS_E" (which
most be defined in sample.voca and represent the silence at the end and at the
start of the sentence). Variables, like we called them previously, are just
another line in the .grammar file which list a sequence of categories or
other variables, and, as you see in the sample file, they can have many
different definitions.

Once you finish playing with the files, you can generate their equivalents in
Julian's native format again, by running the same command as in the quick start:

  mkdfa sample

Then, run julius as showed above and if everything is right it should recognize
your new vocabulary now. In the case you get an error about missing phonemes,
you'll have to remove the words about which it complains, as the available
acoustic models don't support them.

== PRACTICAL UTILITY ==

As you will probably notice, the VoxForge acoustic model is still far away from
being suitable for dictation applications and the like, so its utility right
now is basically restricted to "command and control" applications. For an
example of how to write a simple one in Python, look at the file command.py
in /usr/share/doc/julius-voxforge/examples/controlapp/.

== CONTRIBUTE ==

If you want to help contributing speech corpora for your language, see the
project's main page at http://www.voxforge.org/.

 -- Siegfried-A. Gevatter <rainct@ubuntu.com>. 19/06/2009


debian/changelog
dragonfire (1.1.1) unstable; urgency=medium

  * Upgraded pip version from `9.0.1` to `19.3.1` for TensorFlow `1.15.0`
  * Upgraded TensorFlow version from `1.14.0` to `1.15.0`
  * Upgraded DeepPavlov version from `0.6.1` to `0.7.1`

-- Mehmet Mert Yildiran <mehmetmertyildiran@gmail.com>  Wed, Dec 18 2019 22:15:17 +0300

dragonfire (1.1.0) unstable; urgency=medium

  * The badge generation service changed from Badgen to Shields
  * Omniscient module replaced with DeepPavlov SQuAD BERT and renamed as ODQA
  * Loglevel of TensorFlow set to ERROR and disabled FutureWarning(s)
  * Added a GitHub workflow for automated Docker builds
  * Travis CI is removed
  * `requests` package version upgraded to `>=2.20.0`
  * Post-installation scripts are merged
  * The Docker image has been updated
  * Added separate GitHub workflows to handle Automated Tests and Linter Checks
  * Added a GitHub workflow to test the ODQA module's performance against HotpotQA dataset
  * Fixed most of the major code maintainability issues detected by Code Climate
  * Added a GitHub workflow to automatically publish a GitHub release, build and upload a Debian package to release assets
  * Checksum values are updated in the post-installation script

-- Mehmet Mert Yildiran <mehmetmertyildiran@gmail.com>  Thu, Dec 12 2019 03:43:35 +0300

dragonfire (1.0.4) unstable; urgency=medium

  * Upgraded TensorFlow version from 1.0.0 to 1.14.0
  * Upgraded spaCy version from 2.0.13 to 2.1.3
  * Upgraded NeuralCoref version to 4.0
  * Updated the DeepConversation Tensorflow model and published (v3)
  * Refactored some of the NLP methods
  * Updated the Docker image
  * Added a Makefile to simplify installation commands
  * Fixed some of the tests
  * Fixed the CI pipelines
  * Implemented code coverage by integrating CI to Codecov
  * Integrated Code Climate to have automated code reviews and metrics
  * Fixed Read the Docs documentation build pipeline

 -- Mehmet Mert Yildiran <mehmetmertyildiran@gmail.com>  Fri, Nov 29 2019 01:20:45 +0300

dragonfire (1.0.2) unstable; urgency=medium

  * libgtk2.0-0 and gir1.2-gtk-3.0 dependencies are added
  * The issue related to inactive lock in the speech recognizers is fixed
  * deepspeech Python package dependency is upgraded from 0.2.0a5 to 0.4.1
  * The DeepSpeech the model from 0.1.1 to 0.4.1
  * Time telling feature(a built-in command) is added
  * Several issues in built-in commands are fixed
  * Release version mismatch is fixed

 -- Mehmet Mert Yildiran <mehmetmertyildiran@gmail.com>  Wed, 6 Mar 2019 04:27:54 +0300

dragonfire (1.0.1) unstable; urgency=medium

  * Dependency issues are fixed
  * Several minor bugs are fixed
  * Python 3.6 support added
  * Several linguistic mistakes are fixed
  * A Dockerfile added
  * Command-line formatting improved
  * Add SQLite and MySQL support for the learning ability
  * The knowledge base used in learning, migrated from TinyDB to SQLite/MySQL
  * Module imports are fixed
  * Automatic documentation generation added using Sphinx
  * 134 different tests added and being checked with pytest using Travis CI
  * Coreference resolution added  using NeuralCoref model

 -- Mehmet Mert Yildiran <mehmetmertyildiran@gmail.com>  Mon, 7 Jan 2019 00:26:12 +0300

dragonfire (1.0.0) unstable; urgency=medium

  * Post-installation scripts are fixed
  * An API implemented
  * The project is turned into more of a server-side application although it's continuing to preserve the features of its desktop application aspect
  * The package is now Python 3 only
  * Universal Python wheel generation disabled
  * Basic Analyzer (done in dragonfire/__init__.py) is heavily pruned and improved. Now it's purely spaCy based
  * Omniscient and Learner classes are refactored
  * Learning ability now can store data (that comes from many different Android users) on MySQL database seamlessly
  * Gspeech alternative added

 -- Mehmet Mert Yildiran <mehmetmertyildiran@gmail.com>  Tue, 19 Jun 2018 01:57:10 +0300

dragonfire (0.9.9) unstable; urgency=medium

  * Twitter integration implemented
  * Ability to answer arithmetic questions added
  * A fourth way to respond, Deep Conversation system added

 -- Mehmet Mert Yildiran <mehmetmertyildiran@gmail.com>  Sat, 12 May 2018 14:59:20 +0000

dragonfire (0.9.8) unstable; urgency=medium

  * Migrated from Kaldi Speech Recognition Toolkit to Mozilla DeepSpeech
  * Python3 compatibility improved
  * System tray icon is now GTK+ based instead of wxPython

 -- Mehmet Mert Yildiran <mehmetmertyildiran@gmail.com>  Thu, 26 Apr 2018 21:33:21 +0000

dragonfire (0.9.7) unstable; urgency=medium

  * Python3 support added
  * Omniscient Q&A Engine success rate is improved by ~40%
  * The learning capabilities are improved by including the inverted case
  * Wikipedia connection and disambiguation errors are fixed
  * Several minor bugs are fixed
  * Missing package dependencies are added
  * Post-installation scripts are fixed

 -- Mehmet Mert Yildiran <mehmetmertyildiran@gmail.com>  Sat, 28 Oct 2017 02:39:47 +0000

dragonfire (0.9.6) unstable; urgency=medium

  * Migrated from PyPI SpeechRecognition package to Kaldi Speech Recognition Toolkit
  * Migrated from festival cmu_us_clb_arctic voice to flite slt voice
  * Kaldi GStreamer English model is added
  * Google Search built-in commands are added
  * Several minor bugs are fixed
  * A number of experimental(non-functional) scripts are added

 -- Mehmet Mert Yildiran <mehmetmertyildiran@gmail.com>  Mon, 16 Oct 2017 21:57:32 +0000

dragonfire (0.9.5) unstable; urgency=medium

  * Initial release (Closes: #869121)

 -- Mehmet Mert Yildiran <mehmetmertyildiran@gmail.com>  Mon, 17 Jul 2017 22:20:48 +0000


OLD/controls/.pl
  PID TTY          TIME CMD
    1 ?        00:00:01 init
    2 ?        00:00:00 kthreadd
    3 ?        00:00:01 ksoftirqd/0
    6 ?        00:00:00 migration/0
    7 ?        00:00:00 watchdog/0
    8 ?        00:00:00 migration/1
    9 ?        00:00:10 kworker/1:0
   10 ?        00:00:00 ksoftirqd/1
   12 ?        00:00:00 watchdog/1
   13 ?        00:00:00 migration/2
   15 ?        00:00:00 ksoftirqd/2
   16 ?        00:00:00 watchdog/2
   17 ?        00:00:00 migration/3
   19 ?        00:00:00 ksoftirqd/3
   20 ?        00:00:00 watchdog/3
   21 ?        00:00:00 cpuset
   22 ?        00:00:00 khelper
   23 ?        00:00:00 kdevtmpfs
   24 ?        00:00:00 netns
   26 ?        00:00:00 sync_supers
   27 ?        00:00:00 bdi-default
   28 ?        00:00:00 kintegrityd
   29 ?        00:00:00 kblockd
   30 ?        00:00:00 ata_sff
   31 ?        00:00:00 khubd
   32 ?        00:00:00 md
   34 ?        00:00:00 khungtaskd
   35 ?        00:00:02 kswapd0
   36 ?        00:00:00 ksmd
   37 ?        00:00:00 khugepaged
   38 ?        00:00:00 fsnotify_mark
   39 ?        00:00:00 ecryptfs-kthrea
   40 ?        00:00:00 crypto
   48 ?        00:00:00 kthrotld
   50 ?        00:00:00 scsi_eh_0
   51 ?        00:00:00 scsi_eh_1
   52 ?        00:00:00 kworker/u:2
   53 ?        00:00:00 scsi_eh_2
   54 ?        00:00:00 scsi_eh_3
   55 ?        00:00:01 kworker/u:3
   75 ?        00:00:00 devfreq_wq
  245 ?        00:00:03 jbd2/sda6-8
  246 ?        00:00:00 ext4-dio-unwrit
  270 ?        00:00:01 flush-8:0
  317 ?        00:00:00 upstart-udev-br
  319 ?        00:00:00 udevd
  360 ?        00:00:00 irq/40-mei
  378 ?        00:00:00 cfg80211
  443 ?        00:00:00 udevd
  444 ?        00:00:00 udevd
  484 ?        00:00:00 kpsmoused
  592 ?        00:00:00 upstart-socket-
  621 ?        00:00:00 smbd
  628 ?        00:00:00 smbd
  637 ?        00:00:02 rsyslogd
  638 ?        00:00:02 dbus-daemon
  655 ?        00:00:00 modem-manager
  658 ?        00:00:00 bluetoothd
  662 ?        00:00:00 NetworkManager
  679 ?        00:00:00 avahi-daemon
  681 ?        00:00:00 avahi-daemon
  682 ?        00:00:00 krfcommd
  689 ?        00:00:00 polkitd
  706 ?        00:00:00 cupsd
  716 ?        00:00:00 hd-audio0
  756 ?        00:00:00 hci0
  922 tty4     00:00:00 getty
  929 tty5     00:00:00 getty
  950 tty2     00:00:00 getty
  952 tty3     00:00:00 getty
  953 ?        00:00:00 dictd
  964 tty6     00:00:00 getty
  977 ?        00:00:00 acpid
  990 ?        00:00:02 whoopsie
  993 ?        00:00:05 irqbalance
  996 ?        00:00:00 cron
  998 ?        00:00:00 atd
 1007 ?        00:00:09 mysqld
 1082 ?        00:00:00 mdm
 1083 ?        00:00:16 mdm
 1093 ?        00:06:53 nmbd
 1155 tty8     00:12:25 Xorg
 1269 tty1     00:00:00 getty
 1388 ?        00:00:01 gnome-keyring-d
 1392 ?        00:00:00 console-kit-dae
 1465 ?        00:00:02 gnome-session
 1544 ?        00:00:00 ssh-agent
 1547 ?        00:00:00 dbus-launch
 1548 ?        00:01:50 dbus-daemon
 1562 ?        00:00:31 gnome-settings-
 1568 ?        00:00:02 upowerd
 1663 ?        00:00:01 gvfsd
 1672 ?        00:00:00 gvfs-fuse-daemo
 1701 ?        00:00:00 gsd-printer
 1705 ?        00:21:16 cinnamon
 1707 ?        00:00:00 colord
 1711 ?        00:00:00 accounts-daemon
 1718 ?        00:04:10 pulseaudio
 1720 ?        00:00:00 rtkit-daemon
 1723 ?        00:00:00 gconf-helper
 1725 ?        00:00:00 gconfd-2
 1736 ?        00:00:00 mintupdate-laun
 1738 ?        00:00:00 nm-applet
 1739 ?        00:00:00 polkit-gnome-au
 1744 ?        00:00:00 gnome-fallback-
 1745 ?        00:00:17 syndaemon
 1756 ?        00:00:00 sh
 1757 ?        00:00:01 mintUpdate
 1767 ?        00:00:00 gvfs-gdu-volume
 1770 ?        00:00:00 udisks-daemon
 1771 ?        00:00:00 udisks-daemon
 1801 ?        00:00:00 gvfs-gphoto2-vo
 1803 ?        00:00:00 gvfs-afc-volume
 1843 ?        00:00:00 gvfsd-trash
 1848 ?        00:00:00 gvfsd-burn
 1900 ?        00:01:56 gnome-terminal
 1906 ?        00:00:00 gnome-pty-helpe
 1958 ?        00:00:00 gdu-notificatio
 1999 ?        00:00:04 gnome-screensav
 2000 ?        00:00:01 zeitgeist-datah
 2006 ?        00:00:04 zeitgeist-daemo
 2014 ?        00:00:04 zeitgeist-fts
 2022 ?        00:00:00 cat
 2415 pts/1    00:00:00 bash
 2465 pts/1    00:00:00 sudo
 2466 pts/1    00:00:36 ssh
 2467 ?        00:00:44 nemo
 2474 ?        00:00:00 dconf-service
 2478 ?        00:12:09 chrome
 2484 ?        00:00:09 chrome
 2485 ?        00:00:00 chrome-sandbox
 2486 ?        00:00:00 chrome
 2490 ?        00:00:00 nacl_helper_boo
 2493 ?        00:00:00 gvfsd-metadata
 2494 ?        00:00:00 chrome
 2525 ?        00:01:35 chrome
 2532 ?        00:00:00 chrome
 2536 ?        00:00:03 chrome
 2548 ?        00:05:46 chrome
 2563 ?        00:02:33 chrome
 2572 ?        00:00:27 chrome
 2756 ?        00:02:21 chrome
 2760 ?        00:00:00 chrome
 2965 ?        00:00:00 gvfsd-http
 3185 ?        00:00:23 chrome
 3695 ?        00:00:00 kworker/1:1
 3863 ?        00:00:25 mount.ntfs
 6067 ?        00:10:04 banshee
 7782 ?        00:00:00 dnsmasq
11908 pts/4    00:00:01 bash
13941 ?        00:00:06 chrome
14433 ?        00:02:52 chrome
14819 ?        00:00:03 chrome
16429 pts/5    00:00:00 bash
18478 ?        00:00:06 mount.ntfs
18922 ?        00:00:02 kworker/3:2
19816 ?        00:00:00 kworker/3:0
20187 ?        00:00:51 rhythmbox
21218 pts/4    00:00:16 vi
21268 ?        00:00:07 chrome
21831 ?        00:00:01 kworker/2:2
21832 pts/5    00:00:00 python
21855 pts/3    00:00:00 bash
23853 ?        00:00:00 kworker/0:1
24003 ?        00:00:00 kworker/2:0
24152 ?        00:00:00 kworker/0:0
24450 ?        00:00:29 totem
24570 ?        00:00:00 kworker/2:1
24636 ?        00:00:00 kworker/0:2
24786 pts/3    00:00:03 python
24813 ?        00:00:00 smbd
24846 pts/3    00:00:00 sh
24847 pts/3    00:00:00 ps
